### DISCUSSION TITLE
Why doesn't Anaconda provide an interface for setting a GRUB2 password?

### DISCUSSION THREAD

**Discussion by bitcoffeeiux** [rhinstaller/anaconda#3777] [Ideas]
Grub2 is used as the bootloader for most operating systems. Anaconda also sets grub2 as the boot option for most operating systems. Why does Anaconda not provide a grub2 password set interface? If I have an addon with a grub2 password, can I contribute to the anaconda?

**Comment by jkonecny12** [comment]
I see that there is `--password` kickstart API for the `bootloader` command. Is that sufficient for you.

If you want to have this as GUI option than I think no-one requested that which is probably the main reason. However, I would like to give you a heads-up that we are starting with the new UI initiative https://communityblog.fedoraproject.org/anaconda-is-getting-a-new-suit/ so I'm not sure if you rather don't want to wait with the implementation.

**Reply by bitcoffeeiux** [reply]
I want to be able to integrate this functionality into the GUI, and I'm going to try to write a basic GRUB2 password setting interface contribute to anaconda

**Reply by jkonecny12** [reply]
GRUB 2 password is a bit more advanced and not generally recommended (as I know). We don't want to add everything into the UI for sake of clarity and maintainability.

**Comment by bitcoffeeiux** [comment]


**Reply by jkonecny12** [reply]
Deleting vacation auto-reply comment.
